### PR TITLE
Add external labels to prometheus remote write exporter

### PR DESCRIPTION
**Pull request by amanbrar1999**
**Description:**
This PR adds the concept of Prometheus "external labels" to the Prometheus Remote Write Exporter. These labels must be kept isolated from OTLP metric data point labels because they do not undergo the same type of sanitization. The difference is external labels are allowed to start with "__". This satisfies the use case of Cortex HA deduplication and hence resolves the linked issue.

**Link to tracking Issue:** open-telemetry#1892

**Testing:**
Unit test cases have been added wherever changes were made and manual testing was done to verify Cortex HA deduplication and that nothing changed in metric graphs other than labels.

**Documentation:**
The example configuration in testdata/config.yaml demonstrates how to use the external label config'
README has been updated to include external label config

-- cc @alolita

### CODE REVIEW

File: exporter/prometheusremotewriteexporter/config.go [hunk 1]
```diff
@@ -32,5 +32,13 @@ type Config struct {
 	// See: https://prometheus.io/docs/practices/naming/#metric-names
 	Namespace string `mapstructure:"namespace"`
 
+	ExternalLabels []ExternalLabel `mapstructure:"external_labels"`
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/config.go:35]: Why not having a proper map here for the config?

File: exporter/prometheusremotewriteexporter/helper.go [hunk 2]
```diff
@@ -122,11 +122,25 @@ func timeSeriesSignature(metric *otlp.Metric, labels *[]prompb.Label) string {
...
 	l := map[string]prompb.Label{}
 
+	for _, el := range externalLabels {
+		name := el.Key
+		// The reserved prefix "__" is allowed in an external label
+		if name[:2] == "__" {
+			name = "__" + sanitize(name[2:])
+		} else {
+			name = sanitize(name)
+		}
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/helper.go:137]: Sanitization for external labels can be done in advance and not for every point.

Comment (amanbrar1999) [reply exporter/prometheusremotewriteexporter/helper.go:137]: agreed, will change to to be right after they are scraped

File: exporter/prometheusremotewriteexporter/config.go (hunk 1 above)
Comment (amanbrar1999) [reply exporter/prometheusremotewriteexporter/config.go:35]: i'm not quite sure what you mean, can you clarify?

Comment (amanbrar1999) [reply exporter/prometheusremotewriteexporter/config.go:35]: ah I see, so you mean config being
```
prometheusremotewrite:
  external_labels:
    label1: value1
    label2: value2
```

as opposed to my current implementation of 

```
prometheusremotewrite:
  external_labels:
    - key: label1
       value: value1
    - key: label2
       value: value2
```

Looking back now the first is a lot cleaner so I will change it to that

File: exporter/prometheusremotewriteexporter/factory.go [hunk 3]
```diff
@@ -90,3 +95,23 @@ func createDefaultConfig() configmodels.Exporter {
...
+			key = "__" + sanitize(key[2:])
+		} else {
+			key = sanitize(key)
+		}
+		sanitizedLabels[key] = value
+	}
+
+	cfg.ExternalLabels = sanitizedLabels
+	return nil
+}
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/factory.go:117]: Either change name of the func `validateAndSanitizeExternalLabels` or split this into 2 functions.

**Review by bogdandrutu — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**